python/graphstorm/gconstruct/transform.py

```diff
@@ -52,6 +53,148 @@ def feat_name(self):
+        """
+
+    def collect_info(self, info):
+        """ Store global information for the second phase data processing
+
+        Parameters
+        ----------
+        info:
+            Information to be collected
+        """
+
+class NumericalTransform(TwoPhaseFeatTransform):
+    """ Numerical data Transform ops
+        Do numerical data transformation
+
+    Parameters
+    ----------
+    col_name : str
+        The name of the column that contains the feature.
+    feat_name : str
+        The feature name used in the constructed graph.
+    max_bound : float
+        The maximum float value.
+    min_bound : float
+        The minimum float value
+    """
+    def __init__(self, col_name, feat_name,
+                 max_bound=sys.float_info.max,
+                 min_bound=-sys.float_info.max):
+        self._max_bound = max_bound
+        self._min_bound = min_bound
+        super(NumericalTransform, self).__init__(col_name, feat_name)
+
+    def pre_process(self, feats):
+        """ Pre-process data
+
+        Parameters
+        ----------
+        feats: np.array
+            Data to be processed
+        """
+        assert isinstance(feats, (np.ndarray, HDF5Array)), \
+            "Feature of NumericalTransform must be numpy array or HDF5Array"
+        if isinstance(feats, HDF5Array):
+            feats = feats.to_numpy()
+
+        assert feats.dtype in [np.float64, np.float32, np.float16, np.int64, \
+                              np.int32, np.int16, np.int8], \
+            "Feature of NumericalTransform must be floating points" \
+            "or integers"
+        assert len(feats.shape) <= 2, "Only support 1D fp feature or 2D fp feature"
+        max_val = np.amax(feats, axis=0) if len(feats.shape) == 2 \
+            else np.array([np.amax(feats, axis=0)])
+        min_val = np.amin(feats, axis=0) if len(feats.shape) == 2 \
+            else np.array([np.amin(feats, axis=0)])
+
+        max_val[max_val > self._max_bound] = self._max_bound
+        min_val[min_val < self._min_bound] = self._min_bound
+        return {self.feat_name: (max_val, min_val)}
+
+    def update_info(self, info):
+        """ Store global information for the second phase data processing
+
+        Parameters
+        ----------
+        info: list
+            Information to be collected
+        """
+        max_vals = []
+        min_vals = []
+        for (max_val, min_val) in info:
+            max_vals.append(max_val)
+            min_vals.append(min_val)
+        max_vals = np.stack(max_vals)
+        min_vals = np.stack(min_vals)
+
+        max_val = np.amax(max_vals, axis=0) if len(max_vals.shape) == 2 \
+            else np.array([np.amax(max_vals, axis=0)])
+        min_val = np.amin(min_vals, axis=0) if len(min_vals.shape) == 2 \
+            else np.array([np.amin(min_vals, axis=0)])
+
+        self._max_val = max_val
+        self._min_val = min_val
+
+class NumericalMinMaxTransform(NumericalTransform):
```

Comment:
why this should be a separate class?

Reply:
There can be other Numerical transform ops.